docsk8s/quick-start/ls-k8s-configuration-files.asciidoc

```diff
@@ -0,0 +1,284 @@
+
+
+[float]
+[[qs-single-pipeline-config]]
+==== Single Pipeline
+
+The existing docker image contains a default `pipeline.yml`, which expects a single pipeline, with the definition of that pipeline present in `/usr/share/logstash/pipeline`, as either a single file or collection of files, typically defined as a `ConfigMap` or series of `ConfigMaps` - note that
+a single Kubernetes `ConfigMap` has a size limit of 1MB.
+
+
+This example contains a simple pipeline definition, with the inputs and outputs split into separate configuration files:
+
+
++
+[source,yaml]
+--
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: logstash-pipeline <1>
+  labels:
+    app: logstash-demo
+data:
+  logstash-input.conf: | <2>
+    input {
+      beats {
+        port => "5044"
+      }
+    }
+  logstash-output.conf: | <3>
+    output {
+      elasticsearch {
+        hosts => ["https://demo-es-http:9200"]
+      }
+    }
+--
+
+<1> Name of `ConfigMap` to be referenced in `Deployment`.
+<2> This will create a `ConfigMap` representing the inputs for a pipeline.
+<3> This will create a `CongigMap` representing the outputs for a pipeline.
+
+From there, you will define your `Volume` in your `Deployment` template
+
+[source,yaml]
+--
+volumes:
+  - name: logstash-pipeline
+    configMap:
+      name: logstash-pipeline
+--
+
+and mount the volume in your container
+
+[source,yaml]
+--
+volumeMounts:
+    - name: logstash-pipeline
+      mountPath: /usr/share/logstash/pipeline
+--
+
+
+[float]
+[[qs-multiple-pipeline-config]]
+==== Multiple Pipelines
+
+Multiple pipelines <insert link> require a `ConfigMap` to represent pipelines.yml, where pipeline configurations can be created inline, or in separate `configMap` files or folders.
+
+pipelines.yml `ConfigMap` with an inline pipeline definition
+[source,yaml]
+--
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: logstash-pipeline-yaml <1>
+  labels:
+    app: logstash-demo
+data:
+  pipelines.yml: | <2>
+    - pipeline.id: test
+      pipeline.workers: 1
+      pipeline.batch.size: 1
+      config.string: "input { generator {} } filter { sleep { time => 1 } } output { stdout { codec => dots } }"
+    - pipeline.id: pipeline2 <3>
+      pipeline.workers: 8
+      path.config: "/usr/share/logstash/pipeline2"
+--
+
+Pipelines can also be defined in separate files, as with the single pipeline example
+[source,yaml]
+--
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: pipeline2
+  labels:
+    app: logstash-demo
+data:
+  logstash-input.conf: |
+    input {
+      beats {
+        port => "5044"
+      }
+    }
+  logstash-output.conf: |
+    output {
+      elasticsearch {
+        hosts => ["https://demo-es-http:9200"]
+        index => "kube-apiserver-%{+YYYY.MM.dd}"
+        cacert => "/usr/share/logstash/config/es_ca.crt"
+        user => 'elastic'
+        password => '${ELASTICSEARCH_PASSWORD}'
+      }
+    }
+--
+
+Create the volume in your `Deployment`/`StatefulSet`
```

Comment:
I'm kicking around adding a heading here to visually distinguish this content from previous examples. That might have implications for other sections, so let's discuss.